src/main/index.js

```diff
@@ -1550,6 +1559,15 @@ function runApp() {
     }
   }
 
+  function macOSnewURLWindow(url) {
+    app.once('browser-window-created', (_, mainWindow) => {
+      mainWindow.webContents.once('did-finish-load', () => {
+        // A timeout here is necessary or the new window won't receive the URL
+        setTimeout(function() { mainWindow.webContents.send(IpcChannels.OPEN_URL, url, { isLaunchLink: true }) }, 1000)
+      })
+    })
+  }
+
```

Comment:
Would 

```js
  async function macOSnewURLWindow(url) {
    app.once('browser-window-created', async (_, mainWindow) => {
      await mainWindow.webContents.once('did-finish-load', () => {
        mainWindow.webContents.send(IpcChannels.OPEN_URL, url, { isLaunchLink: true })
      })
    })
  }
```

work better, instead of timeout? Are we just waiting for a promise to resolve?

Reply:
Nope, as you would be sending the URL as soon as the HTML has loaded but the JavaScript is loaded asynchronously and it still needs to load all of your user data such as the settings and actually register the handler for that IPC call. Please see my better solution below, which is the same as what is done when the app originally starts, which is waiting for the window to say it is ready (refer to `src/renderer/App.js` for the window-side code.